### DISCUSSION TITLE
Skipping Chunk due to ownership

### DISCUSSION THREAD

**Discussion by JV73** [polarfire-soc/polarfire-soc-documentation#330] [HSS and U-Boot]
The HSS is rejecting my AMP package during boot.
![Capture](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/106761965/29d476a1-32f0-4f04-8a33-806c27163e12)
Where did we go wrong ?

**Comment by griffini** [comment]
Hi @JV73, would you mind sharing a little more information to help us investigate:

* What HSS version is this with?
* Could you share the YAML file you used for the HSS payload generator?
* What payloads are you booting? Linux and a bare metal?
* Which AMP context isn't booting? (e.g. the Linux side, or the bare metal side?)
* Did you try the SKIP_OPENSBI flag for the bare metal side?

**Reply by JV73** [reply]
There is a uboot/linux build on Harts1-3 that is booting successfully, and a FreeRTOS application on HART4 that is being skipped.  This same 'package' was created for the ICICLE Kit and boots successfully.  Now it has been 'ported' to new hardware targeting an MPFS095 implementation.  I'll have to check with the team to see which MSS_CFG, HSS, Libero versions are being used - stay tuned.

**Reply by Shivaraj199** [reply]
hey @JV73  I am having this same error, the payload works on the Icicle kit but not on the videokit . Were you able to solve this?

**Reply by jgonzalo** [reply]
same here, any solution?
Merci
Tx

**Reply by vfalanis** [reply]
Hi,

Are you trying to port the polarfire-soc-amp-examples to another device or to the video kit? If so, do you have the Mi-V IHC IP instantiated in the fabric?
Please note that the polarfire-soc-amp-example FreeRTOS example project communicates between the two AMP contexts. Therefore, in addition to porting the example to your device, you also need to have the Mi-V IHC IP present in the fabric.

**Reply by jgonzalo** [reply]
Thank you, Yes, AMP in video-kit is my goal. I have learnt these days about the need of IHC IP (missing in my current implementation). So there seem to be two ways ahead:
- Use shared memory (sucessful) and soft interrupts to notify updated data: how can I catch soft interrupts in Linux? any other notification mechanism?
- Use the same mechanism as HSS to exchange messages among harts: does not it use OpenSBI or IHC?

Thank you again for your response
Jesús
